File: 2-Patron/python/MascotaPequeña.py
from _typeshed import Self
import Criatura


class MascotaPequeña(Criatura):
    def __init__(self, nombre, defensa, ataque, nivel):
        super().__init__(nombre, defensa, ataque, nivel)
        self.__sígueme = True

    @property
    def sígueme(self):
        return self.__sígueme

    @sígueme.setter
    def sígueme(self, new):
        self.__sígueme = new

    def pasear(self):
        self.__sígueme = True
        return self.__sígueme

    def sentar(self):
        self.__sígueme = False
        return self.__sígueme

    def decSalud(self, decremento):
        self.__salud -= decremento // self.__defensa
        return self.__salud

    def incNivel(self):
        self.__nivel += 5
        return self.__nivel

    def verStats(self):
        return f"{super().verStats()}\n\
                 Sígueme: {self.__sígueme}\n"

File: 2-Patron/Java/MascotaPequeña.java
package abstractmasterwhole;
public class MascotaPequeña extends Criatura{
    private boolean sigueme;
    public MascotaPequeña(String nombre, int defensa, int ataque, int nivel) {
        super(nombre, defensa, ataque, nivel);
        this.sigueme = true;
    }
    public boolean getSigueme(){
        return this.sigueme;
    }
    public void setSigueme(boolean sigueme){
        this.sigueme = sigueme;
    }
    public boolean pasear(){
        this.sigueme = true;
        return sigueme;
    }
    public boolean sentar(){
        this.sigueme = false;
        return sigueme;
    }

    @Override
    public int decSalud(int dec) {
        
        return this.salud = decremento / this.defensa;
    }

    @Override
    public int incNivel() {
        return this.nivel +=5;
    }

    @Override
    public String verStats() {
        return super.verStats() + "Siguieme: "+this.sigueme; //To change body of generated methods, choose Tools | Templates.
    }

    
    
    
    
}
